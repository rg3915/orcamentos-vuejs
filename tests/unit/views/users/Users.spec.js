import { createLocalVue, shallowMount } from '@vue/test-utils'
import CoreuiVue from '@coreui/vue'
import Users from '@/views/users/Users'
import VueRouter from 'vue-router';

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = new VueRouter()

localVue.use(CoreuiVue)

describe('Users.vue', () => {
  it('has a name', () => {
    expect(Users.name).toBe('Users')
  })
  it('has a created hook', () => {
    expect(typeof Users.data).toMatch('function')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.vm).toBeTruthy()
  })
  it('is Users', () => {
    const wrapper = shallowMount(Users,{
      localVue,
      router
    })
    expect(wrapper.findComponent(Users)).toBeTruthy()
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Users, {
      localVue,
      router
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
