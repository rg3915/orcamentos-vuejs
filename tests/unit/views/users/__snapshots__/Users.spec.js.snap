// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users.vue renders correctly 1`] = `
<crow-stub
  gutters="true"
  tag="div"
>
  <ccol-stub
    col="12"
    tag="div"
    xl="8"
  >
    <ccard-stub>
      <ccardheader-stub>
        
        Users
      
      </ccardheader-stub>
       
      <ccardbody-stub>
        <cdatatable-stub
          activepage="1"
          clickablerows="true"
          fields="[object Object],[object Object],[object Object],[object Object]"
          header="true"
          hover="true"
          items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
          itemsperpage="5"
          pagination="[object Object]"
          responsive="true"
          sortervalue="[object Object]"
          striped="true"
        />
      </ccardbody-stub>
    </ccard-stub>
  </ccol-stub>
</crow-stub>
`;
